x/launch/keeper/validator_set.go

```diff
@@ -0,0 +1,64 @@
+package keeper
+
+import (
+	"bytes"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/tendermint/spn/x/launch/types"
+	tmtypes "github.com/tendermint/tendermint/types"
+)
+
+// CheckValidatorSet check the information about a validator
+// set (used to create an IBC client) are valid
+func (k Keeper) CheckValidatorSet(
+	ctx sdk.Context,
+	launchID uint64,
+	chainID string,
+	validatorSet tmtypes.ValidatorSet,
+) (bool, error) {
+	// check chain ID
+	chain, found := k.GetChain(ctx, launchID)
+	if !found {
+		return false, sdkerrors.Wrapf(types.ErrChainNotFound, "%d", launchID)
+	}
+	if !chain.LaunchTriggered {
+		return false, sdkerrors.Wrapf(types.ErrNotTriggeredLaunch, "%d", launchID)
+	}
+	if chain.GenesisChainID != chainID {
+		return false, sdkerrors.Wrap(types.ErrInvalidGenesisChainID, chainID)
+	}
+
+	// all validators must be present in the launch module and
+	// the total amount of self-delegation from the provided validators
+	// must reach at least 2/3 of the total self delegation for the chain
+	valSetSelfDelegation := 0.0
+	for _, validator := range validatorSet.Validators {
+		valAddr := sdk.AccAddress(validator.Address.Bytes())
+		launchValidator, found := k.GetGenesisValidator(ctx, launchID, valAddr.String())
+		if !found {
+			return false, sdkerrors.Wrap(types.ErrValidatorNotFound, valAddr.String())
+		}
+		valConsPubKey := validator.PubKey.Bytes()
+		if bytes.Compare(launchValidator.ConsPubKey, valConsPubKey) != 0 { // nolint
+			return false, sdkerrors.Wrap(types.ErrInvalidConsPubKey, validator.PubKey.Address().String())
+		}
+		valSetSelfDelegation += float64(launchValidator.SelfDelegation.Amount.Int64())
```

Comment:
I think we should do the decimal conversion at the last check and keep integer here